Sources/NIO/System.swift

```diff
@@ -99,14 +99,18 @@ internal func wrapSyscallMayBlock<T: FixedWidthInteger>(where function: StaticSt
     }
 }
 
+internal func continueOnEINTR(errnoCode: CInt) -> Bool {
+    return errnoCode == EINTR
+}
+
 /* Sorry, we really try hard to not use underscored attributes. In this case however we seem to break the inlining threshold which makes a system call take twice the time, ie. we need this exception. */
 @inline(__always)
-internal func wrapSyscall<T: FixedWidthInteger>(where function: StaticString = #function, _ body: () throws -> T) throws -> T {
+internal func wrapSyscall<T: FixedWidthInteger>(where function: StaticString = #function, continueOnErrno: (CInt) -> Bool = continueOnEINTR, _ body: () throws -> T) throws -> T {
```

Comment:
Should we just make this a separate function rather than use the closure?

Reply:
@Lukasa I am not strong on it... that said I think the closure should be free so why not just re-use the rest here ? @weissi WDYT ?

Reply:
I'm a bit reluctant to combine `@inline(__always)` with this closure, that's all.

Reply:
ok let me just add an extra method then... Its easy enough to do :)